File: FrontEnd/src/service/StudentService.js
import axios from "axios";

const BASE_URL = "http://localhost:8080/StudentApp/v1";


class StudentService {

  
  addStudent(student){
    return axios.post(BASE_URL + '/addStudent', student);
  }

  getStudents(){
    return axios.get(BASE_URL + '/students');
  }

  getStudentByName(name){
    return axios.get(BASE_URL + '/student', {
        params: {name: name}
    })
  }

  updateStudent(student){
    return axios.put(BASE_URL + '/update/student', student);
  }

  deleteStudent(name){
    return axios.delete(BASE_URL + '/delete/student', {
        params: {name: name}
    })
  }


}


export default new StudentService();

File: BackEnd/src/main/java/com/syfassignment/studentmanagementapp/service/StudentService.java
package com.syfassignment.studentmanagementapp.service;

import com.syfassignment.studentmanagementapp.entity.Student;
import java.util.List;

public interface StudentService {

    //API to get all the Students data
    List<Student> getAllStudents();

    //API to get a student by his name if his data exists
    Student getStudentByName(String name);

    //API to get a student by his ID if his data exists
    Student getUserById(int id);

    //API to add a new student
    Student addStudent(Student student);

    //API to update the student data if exists
    String updateStudent(Student student);

    String deleteStudent(String name);
}
